cmd/nerdctl/compose_exec_linux_test.go

```diff
@@ -207,3 +211,83 @@ services:
 	base.ComposeCmdWithHelper(unbuffer, "-f", comp.YAMLFullPath(), "exec", "-t=false", "svc0", "stty").AssertFail()                         // `-i`
 	base.ComposeCmdWithHelper(unbuffer, "-f", comp.YAMLFullPath(), "exec", "-i=false", "-t=false", "svc0", "stty").AssertFail()
 }
+
+func TestComposeExecWithIndex(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	var dockerComposeYAML = fmt.Sprintf(`
+version: '3.1'
+
+services:
+  svc0:
+    image: %s
+    command: "sleep infinity"
+    deploy:
+      replicas: 3
+`, testutil.CommonImage)
+
+	comp := testutil.NewComposeDir(t, dockerComposeYAML)
+	t.Cleanup(func() {
+		comp.CleanUp()
+	})
+	projectName := comp.ProjectName()
+	t.Logf("projectName=%q", projectName)
+
+	base.ComposeCmd("-f", comp.YAMLFullPath(), "up", "-d", "svc0").AssertOK()
+	t.Cleanup(func() {
+		base.ComposeCmd("-f", comp.YAMLFullPath(), "down", "-v").AssertOK()
+	})
+
+	// try random 5~10 times to ensure that results are stable
+	r, _ := rand.Int(rand.Reader, big.NewInt(5))
+	for i := 0; i < int(r.Int64()+5); i++ {
+		for _, i := range []string{"1", "2", "3"} {
+			host := fmt.Sprintf("%s-svc0-%s.%s_default", projectName, i, projectName)
+			var (
+				expectIP string
+				realIP   string
+			)
+			cmds := []string{"-f", comp.YAMLFullPath(), "exec", "-i=false", "-t=false", "--index", i, "svc0"}
+			base.ComposeCmd(append(cmds, "cat", "/etc/hosts")...).
+				AssertOutWithFunc(func(stdout string) error {
+					lines := strings.Split(stdout, "\n")
+					for _, line := range lines {
+						if !strings.HasSuffix(line, host) {
+							continue
+						}
+						fields := strings.Fields(line)
+						expectIP = fields[0]
+						return nil
+					}
+					return errors.New("fail to get the expected ip address")
+				})
+			base.ComposeCmd(append(cmds, "ip", "addr", "show", "dev", "eth0")...).
+				AssertOutWithFunc(func(stdout string) error {
+					ip := findIP(stdout)
+					if ip == nil {
+						return errors.New("fail to get the real ip address")
+					}
+					realIP = ip.String()
+					return nil
+				})
+			assert.Equal(t, realIP, expectIP)
+		}
+	}
+}
+
+func findIP(output string) net.IP {
+	var ip string
+	lines := strings.Split(output, "\n")
+	for _, line := range lines {
+		if !strings.Contains(line, "inet ") {
+			continue
+		}
+		fields := strings.Fields(line)
+		if len(fields) <= 1 {
+			continue
+		}
```

Comment:
should we do the same check in line 259?

Reply:
make sense